### ISSUE TITLE
Fieldset issue - Legend not read by NVDA

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#5291]
**Reported by Bogdana on 2015-08-18 05:11**
There is a specific case, where NVDA is not reading the `<legend>`:

Lets say we have a group of form elements surrounded by `<fieldset>` in combination with `<legend>`. When tabbing through the page, the focus lands on the first form element and the reader is supposed to say whatever we have in the legend + grouping at the end. Ok, but that's _not the case if the first form element is an application and our focus before that was on a  link_. If the previously focused element was a native form element it works just fine.

_Not working:_
```
<a href="#">Link 1</a>
  <fieldset>
    <legend>Fieldset legend</legend>
      <div role="application">
        <a href="#">Link 2</a>
      </div>
  </fieldset>
```
_Working:_
```
<input type="text" />
  <fieldset>
     <legend>Fieldset legend</legend>
        <div role="application">
          <a href="#">Link 2</a>
        </div>
  </fieldset>
```
Any thoughts on the matter ?

**Comment by kymdusting** [comment]
Also running into this problem although the circumstances are a little more specific.

The scenario is:
The fieldset contains the first set of form elements in the page.
The first element to get focus within that fieldset is a button or link (not any kind of input).
Tabbing through the fieldset the legend is ignored and does not read "legend text grouping" on focus of any elements with the fieldset.
If you reverse tab through the fieldset so that the first element to get focus is an input it does read "legend text grouping".

This only happens if the fieldset is the first element in the page containing form elements. The same fieldset moved after another form field will read the "legend text grouping" correctly.

**Comment by feerrenrut** [comment]
I believe this will be fixed with #3321. @jcsteh Can you confirm?

**Comment by jcsteh** [comment]
Yes, #3321 should fix this. Closing as duplicate.
